source/common/network/utility.h

```diff
@@ -16,21 +16,48 @@ namespace Network {
  * Utility class for keeping a list of IPV4 addresses and masks, and then determining whether an
  * IP address is in the address/mask list.
  */
-class IpWhiteList {
+class IpList {
 public:
-  IpWhiteList(const Json::Object& config);
+  IpList(const std::vector<std::string>& subnets);
+  IpList(){};
 
-  bool contains(const std::string& remote_address) const;
+  bool contains(const std::string& address) const;
+  bool empty() const { return ipv4_list_.empty(); }
 
 private:
   struct Ipv4Entry {
     uint32_t ipv4_address_;
     uint32_t ipv4_mask_;
   };
 
-  std::vector<Ipv4Entry> ipv4_white_list_;
+  std::vector<Ipv4Entry> ipv4_list_;
+};
+
+class IpWhiteList {
```

Comment:
Can we just get rid of this class and just have IpList have a constructor that accepts JSON?